logstash-core/lib/logstash/api/lib/app/commands/stats/memory_command.rb

```diff
@@ -0,0 +1,21 @@
+# encoding: utf-8
+require "app/command"
+require 'monitoring'
+
+class LogStash::Api::JvmMemoryCommand < LogStash::Api::Command
+
+  def run
+    memory = LogStash::Json.load(service.get(:jvm_memory_stats))
+    {
+      :heap => memory["heap"],
+      :non_heap => memory["non_heap"],
+      :pools => memory["pools"].inject({}) do |acc, (type, hash)|
+                  acc[type] = hash
+                  acc
+               end
+    }
+  rescue
+    {} # Something happen, so we just return an empty hash.
+  end
```

Comment:
This rescue block is rescuing everything, what are the actuals exceptions you are thinking that can happen here?